build/templates/_attributes.py.mako

```diff
@@ -134,7 +134,10 @@ class AttributeEnumWithConverter(AttributeEnum):
             raise errors.DriverTooNewError('The driver runtime returned an unexpected value. ')
 
     def __set__(self, session, value):
-        return self._underlying_attribute_enum.__set__(session, self._setter_converter(value))
+        try:
+            return self._underlying_attribute_enum.__set__(session, self._setter_converter(value))
+        except KeyError:
+            raise ValueError(f'{value} cannot be converted to an {str(self._underlying_attribute_enum._attribute_type.__name__)} enum value')
```

Comment:
Imagine you are a Python using customer.
You set an invalid value of 3.

The fact that the 3 gets converted under the hood to some private enum is an irrelevant (to the customer) implementation detail. What are they supposed to do with that information?

Let's make this simple, useful, and not confusing.

> Invalid value: 3